### PR TITLE
Add yarn

### PR DESCRIPTION
**Pull request by sunderls**

### CODE REVIEW

File: lib/linguist/generated.rb [hunk 1]
```diff
@@ -58,6 +58,7 @@ def generated?
       node_modules? ||
       go_vendor? ||
       npm_shrinkwrap? ||
+      yarn_lock? ||
```
Comment (pchaigno) [inline lib/linguist/generated.rb]: Could you move this to the end of the list instead?
This list of functions starts with the functions that do not need to load the file (only use the extension and path).

Comment (sunderls) [inline lib/linguist/generated.rb]: @pchaigno  sorry for lack of ruby experience, could you help me with this error? 🙇 
```
1) Error:
TestFileBlob#test_generated:
Errno::ENOENT: No such file or directory - /home/travis/build/github/linguist/samples/Gemfile.lock
    /home/travis/build/github/linguist/lib/linguist/file_blob.rb:33:in `read'
    /home/travis/build/github/linguist/lib/linguist/file_blob.rb:33:in `data'
    /home/travis/build/github/linguist/lib/linguist/blob_helper.rb:319:in `block in generated?'
```

**Comment by sunderls** [comment]
@pchaigno  sorry for lack of ruby experience, could you help me with this error? 🙇 
```
1) Error:
TestFileBlob#test_generated:
Errno::ENOENT: No such file or directory - /home/travis/build/github/linguist/samples/Gemfile.lock
    /home/travis/build/github/linguist/lib/linguist/file_blob.rb:33:in `read'
    /home/travis/build/github/linguist/lib/linguist/file_blob.rb:33:in `data'
    /home/travis/build/github/linguist/lib/linguist/blob_helper.rb:319:in `block in generated?'
```

**Comment by pchaigno** [comment]
Do these generated files always have the same filename (i.e., `yarn.lock`)?

**Comment by wopian** [comment]
Yeah, it's [always](https://yarnpkg.com/en/docs/yarn-lock) `yarn.lock`

**Comment by pchaigno** [comment]
Great Thanks @wopian!

@sunderls Could you match on the string `# THIS IS AN AUTOGENERATED FILE` and check that the filename is `yarn.lock` (`name.match(/yarn\.lock/)`)?
That should also fix the error.

**Comment by sunderls** [comment]
@wopian @pchaigno  Many Thanx.

Both filename & first line matching are applied.

**Pull request merged**